### DISCUSSION TITLE
Is it possible to visualize class diagrams in my favorite IDE (PyCharm, vscode, etc)?

### DISCUSSION THREAD

**Discussion by cmungall** [linkml/linkml#2240] [Q&A]
The LinkML docs provide instructions on setting up syntax checking for LinkML schemas in IDEs. (and using copilot with schemas works really well!).

Is it also possible to do any kind of navigation of the classes directly in the IDE, or visualize?

**Comment by sierra-moxon** [comment]
For folks using PyCharm (the community edition is free and works well with LinkML), I've found that after generating a PlantUML `.puml` file with LinkML's `gen-plantuml` command, visualizing it in the IDE itself is especially easy via a free PyCharm plugin.

Steps:
- PyCharm (https://www.jetbrains.com/pycharm/download/download-thanks.html) 
- Install the PlantUML plugin via `Preferences->Plugins` (https://plugins.jetbrains.com/plugin/7017-plantuml-integration)
- Open your linkml project in PyCharm (generated via the[ linkml-project-cookiecutter](https://github.com/linkml/linkml-project-cookiectutter))

<img width="1585" alt="Screen Shot 2024-07-31 at 9 46 28 AM" src="https://github.com/user-attachments/assets/b759a638-564c-4a98-a4cd-098c1db72b9a">

- Generate the PlantUML diagram using your schema
```
% poetry install
% poetry run gen-plantuml src/[your_project_name]/schema/[your_project_name].yaml > project/plantuml/schema.puml
```

Back in Pycharm, navigate to the `project/plantuml/schema.puml` file and view it!

<img width="1804" alt="Screen Shot 2024-07-31 at 9 44 22 AM" src="https://github.com/user-attachments/assets/60ed18fc-ce91-43bb-a85c-637c16f19736">
